plugins/home/src/api/LocalStorageVisitsApi.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Visit } from './VisitsApi';
+import { VisitsApiFactory } from './VisitsApiFactory';
+
+export class LocalStorageVisitsApi extends VisitsApiFactory {
+  private readonly localStorage: Window['localStorage'];
```

Comment:
Hm. Should this perhaps use [`storageApiRef`](https://github.com/backstage/backstage/blob/723ef86686f4029e933c1c2a2346c07df89cb3d7/packages/core-plugin-api/src/apis/definitions/StorageApi.ts#L108)? That would allow users to even back it with the user-settings backend if they want, getting cross-client visits tracked.

That means we'd probably want to heavily debounce the storage calls to it, so we don't generate tons of network traffic for every visit. But it might be beneficial?

Of course there's nothing stopping us from BOTH having an explicit always-local-storage implementation, as well as a storage-ref-based implementation to pick among.

Reply:
That is a good idea!
My intention here was to create a reference implementation for the moment backed by localStorage to be used as a starting point.
Do you mind if I work on storageApi implementation on a following PR?

Reply:
Scratch that. I've added a new commit with an implementation backed by the StorageApi 🎉 . Thanks again for the idea.